datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -346,14 +346,18 @@ fn optimize(plan: &LogicalPlan, mut state: State) -> Result<LogicalPlan> {
                 .fields()
                 .iter()
                 .enumerate()
-                .map(|(i, field)| {
+                .flat_map(|(i, field)| {
                     // strip alias, as they should not be part of filters
                     let expr = match &expr[i] {
                         Expr::Alias(expr, _) => expr.as_ref().clone(),
                         expr => expr.clone(),
                     };
 
-                    (field.qualified_name(), expr)
+                    // considering both qualified and unqualified names as rewriting option
+                    [
+                        (field.name().clone(), expr.clone()),
+                        (field.qualified_name(), expr),
+                    ]
```

Comment:
Why is this change to `Projection` needed? I tried removing this change locally and all the tests still passed 🤔 

Can you add a test case that would fail without this change?

Reply:
Yes, that change is not really necessary, i guess - all downstream operations manages to resolve proper objects for forwarding filters (at least I checked both simple projections and joins). It just looked more consistent to support unqualified columns resolution.

I can revert this change if other parts look fine, np.

Reply:
Done, waiting for CI

Thank you!